File: app/src/androidTest/java/net/tlalka/android/fiszki/test/ActivityLazyRule.java
package net.tlalka.android.fiszki.test;

import android.app.Activity;
import android.support.test.rule.ActivityTestRule;

public class ActivityLazyRule<T extends Activity> extends ActivityTestRule<T> {

    public ActivityLazyRule(Class<T> activityClass) {
        super(activityClass, true, false);
    }

    public T launchActivity() {
        return super.launchActivity(super.getActivityIntent());
    }
}

File: app/src/androidTest/kotlin/net/tlalka/fiszki/test/ActivityLazyRule.kt
package net.tlalka.fiszki.test

import android.app.Activity
import android.support.test.rule.ActivityTestRule

class ActivityLazyRule<T : Activity>(activityClass: Class<T>) : ActivityTestRule<T>(activityClass, true, false) {

    fun launchActivity(): T {
        return super.launchActivity(super.getActivityIntent())
    }
}
